Mathlib/Analysis/Convex/Birkhoff.lean

```diff
@@ -169,4 +171,29 @@ theorem doublyStochastic_eq_convexHull_permMatrix :
     obtain ⟨w, hw1, hw2, hw3⟩ := exists_eq_sum_perm_of_mem_doublyStochastic hM
     exact mem_convexHull_of_exists_fintype w (·.permMatrix R) hw1 hw2 (by simp) hw3
 
+/--
+The set of extreme points of the doubly stochastic matrices is the set of permutation matrices.
+-/
+theorem extremePoints_doublyStochastic :
+    Set.extremePoints R (doublyStochastic R n) = {σ.permMatrix R | σ : Equiv.Perm n} := by
+  refine subset_antisymm ?_ ?_
+  · rw [doublyStochastic_eq_convexHull_permMatrix]
+    exact extremePoints_convexHull_subset
+  rintro _ ⟨σ, rfl⟩
+  refine ⟨permMatrix_mem_doublyStochastic, fun x₁ hx₁ x₂ hx₂ hσ ↦ ?_⟩
+  suffices ∀ i j : n, x₁ i j = x₂ i j by
+    obtain rfl : x₁ = x₂ := by simpa [← Matrix.ext_iff]
+    simp_all
```

Comment:
I think the definition of `Set.extremePoints` is bad. It shouldn't end with `... x₁ = x ∧ x₂ = x`, but rather `... x₁ = x₂`. If someone wants to prove they are both `x`, then they can just use `trans x`.

Maybe that's something that should be fixed in a different PR, but I thought I would point it out.

Reply:
Writing this proof certainly made me feel the same way... But I agree that this PR probably isn't the place to fix it. Do you mind if I leave it as-is for now?